### PR TITLE
Raids

### PR DESCRIPTION
**Pull request by otooleam**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ## Discord yells at us if this file gets uploaded
-token.txt
+env.json
```
Comment (Marthyu-Verat) [inline .gitignore:2]: This file should not be ignored as it is required to be filled out for the bot to run. You should only check in this file if you change the keys inside of it, otherwise discard the changes before pushing.

File: PokeStar/PokeStar/ConnectionInterface/DatabaseConnector.cs [hunk 2]
```diff
@@ -90,6 +90,24 @@ FROM pokemon
          return pokemon;
       }
 
+      public List<string> GetBossList(short tier)
+      {
+         List<string> bosses = new List<string>();
+         using (var conn = GetConnection())
+         {
+            conn.Open();
+            string queryString = $"SELECT pokemon FROM raid_boss WHERE tier = {tier} AND active = 1";
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/ConnectionInterface/DatabaseConnector.cs]: This causes uneeded updates to the db were reading from sliph would be more efficient

File: PokeStar/PokeStar/ConnectionInterface/SilphData.cs [hunk 3]
```diff
@@ -1,77 +1,77 @@
-﻿using System;
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/ConnectionInterface/SilphData.cs]: If file is fully commented out then should be removed not just commented out

File: PokeStar/PokeStar/Modules/DexCommands.cs [hunk 4]
```diff
@@ -86,7 +86,7 @@ public async Task CP([Remainder] string text)
 
             await Context.Channel.SendFileAsync(fileName, embed: embed.Build()).ConfigureAwait(false);
 
-            Connections.CopyFile(fileName);
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/Modules/DexCommands.cs:89]: Why was this removed, is needed for pictures

File: PokeStar/PokeStar/Program.cs [hunk 5]
```diff
@@ -103,10 +105,12 @@ public void HookReactionAdded()
       {
          var message = await cachedMessage.GetOrDownloadAsync().ConfigureAwait(false);
          var user = reaction.User.Value;
-         if (message != null && reaction.User.IsSpecified && !user.IsBot)
-            Console.WriteLine($"{reaction.User.Value} just added a reaction '{reaction.Emote}' " +
-                              $"to {message.Author}'s message ({message.Id}).");
+         if (message != null && reaction.User.IsSpecified && !user.IsBot && RaidCommand.IsCurrentRaid(message.Id))
+         {
+            //Console.WriteLine($"{reaction.User.Value} just added a reaction '{reaction.Emote}' to {message.Author}'s message ({message.Id}).");
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/Program.cs]: remove commented out line

File: PokeStar/PokeStar/Program.cs [hunk 6]
```diff
@@ -54,8 +58,6 @@ public async Task MainAsync()
          Environment.SetEnvironmentVariable("SETUP_TRADE", "FALSE");
          Environment.SetEnvironmentVariable("SETUP_DEX", "FALSE");
 
-         var connectors = Connections.Instance();
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/Program.cs]: line should not be removed

File: PokeStar/env.json [hunk 7]
```diff
@@ -1,5 +0,0 @@
-{
```
Comment (Marthyu-Verat) [inline PokeStar/env.json:1]: File should not be removed

**Comment by Marthyu-Verat** [comment]
If you added or changed the Database please remember to commit the backed up database file

File: PokeStar/PokeStar/Modules/DexCommands.cs (hunk 4 above)
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/Modules/DexCommands.cs:89]: Should be changed to DeleteFile

File: PokeStar/PokeStar/DataModels/RaidBoss.cs [hunk 8]
```diff
@@ -20,5 +20,26 @@ public class RaidBoss
       public int CPHigh { get; set; }
       public int CPLowBoosted { get; set; }
       public int CPHighBoosted { get; set; }
+
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/DataModels/RaidBoss.cs:23]: Remove unused code

File: PokeStar/PokeStar/DataModels/Raid.cs [hunk 9]
```diff
@@ -1,40 +1,110 @@
-﻿using System;
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/DataModels/Raid.cs:1]: Remove unused using statements

File: PokeStar/PokeStar/DataModels/Raid.cs [hunk 10]
```diff
@@ -1,40 +1,110 @@
...
+         }
+         else if (Here.ContainsKey(player))
+         {
+            PlayerCount -= Here[player];
+            HereCount -= Here[player];
+            Here.Remove(player);
+         }
+      }
+
+      private List<string> GetBossData(short tier)
```
Comment (Marthyu-Verat) [inline PokeStar/PokeStar/DataModels/Raid.cs:99]: Remove unused method

**Review by Marthyu-Verat — COMMENTED** [review]
We should discuss design of the db

File: PokeStar/PokeStar/Modules/DexCommands.cs (hunk 4 above)
Comment (otooleam) [reply PokeStar/PokeStar/Modules/DexCommands.cs:89]: You wrote it that way and I didn't really understand it but removing it didn't break anything. I will change it to DeleteFile
